### DISCUSSION TITLE
Low performance in wiring mode in sub editor

### DISCUSSION THREAD

**Discussion by PlsLetMeThrough** [FakeFishGames/Barotrauma#13951] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

While using wiring mode in sub editor fps drops drasticly, however if you hover over any interface element the fps rises. You can even move the cursor or the camera using wasd while hovering over the interface and it doesn't matter. I know that bigger ships will always cause performance issues, but while in main editor fps stays above 15, in wiring mode it may drop below 5 if you don't hover over interface. Every interface element has the same effect.
![image1](https://github.com/FakeFishGames/Barotrauma/assets/155765522/6a1e2159-8e3a-4bec-9914-df51af3584b6)
![image2](https://github.com/FakeFishGames/Barotrauma/assets/155765522/ae1b6adc-6877-4f5e-b5ec-d20cd53fd895)
![image3](https://github.com/FakeFishGames/Barotrauma/assets/155765522/fa0db15b-3c4b-4798-9f85-16536202fb51)

### Reproduction steps

1. Open sub editor
2. Load big enough sub
3. Go into wiring mode
4. Observe the fps drop
5. Hover over interface element
6. Observe the fps rise

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Other

### -

Sub editor

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
- Have you tried disabling wiring mode transparency in shift+RMB menu?
- Omg, dude, use less items, use custom textures instead of decorations, use circuit boxes, create walls with custom shapes if you need, have mercy on your cpu and cpus of your crewmates

**Reply by PlsLetMeThrough** [reply]
1. Transparency doesn't change a thing.
2. The sub is only 60% done and is still playable although on <30 fps. BTW cpu is chillin on 50%. I know that subs this big are very problematic, but I made this report due to finding out that hovering over interface elements increases fps, although it doesn't really help cause you can't wire while doing that. Maybe something can be done with this info idk.
